core/shared/src/main/scala/zio/Hub.scala

```diff
@@ -229,6 +229,7 @@ object Hub {
           ZIO
             .whenZIO(shutdownHook.succeed(())) {
               ZIO.foreachPar(unsafePollAll(pollers))(_.interruptAs(fiberId)) *>
+                ZIO.succeed(subscribers.remove(subscription -> pollers)) *>
                 ZIO.succeed(subscription.unsubscribe()) *>
                 ZIO.succeed(strategy.unsafeOnHubEmptySpace(hub, subscribers))
```

Comment:
What do you think about:
```suggestion
                ZIO.succeed {
                  subscribers.remove(subscription -> pollers))
                  subscription.unsubscribe()
                  strategy.unsafeOnHubEmptySpace(hub, subscribers)
                }
```
?

Reply:
I find the way it is currently written more readable and feels like a premature optimization gives that this code is only run when the subscription is finalized but you can do that too!